src/mpid/common/shm/mpidu_init_shm.h

```diff
@@ -12,6 +12,11 @@
 /* One cache line per process should be enough for all cases */
 #define MPIDU_INIT_SHM_BLOCK_SIZE MPIDU_SHM_CACHE_LINE_LEN
 
+#if defined(MPL_USE_LOCK_BASED_PRIMITIVES)
```

Comment:
Thank you very much for this commit and `--enable-nolocal` PR!
Since we have not completely moved to MPL/atomics, could you add `|| defined(OPA_USE_LOCK_BASE_PRIMITIVES)` as well? Unlike the current MPL/atomics, this flag is enabled only when an emulation lock is used.

Reply:
Thanks. I got it. Then `OPA_USE_LOCK_BASE_PRIMITIVES` is unnecessary.